src/bauhaus/bauhaus.c

```diff
@@ -504,25 +502,31 @@ void dt_bauhaus_init()
   gtk_style_context_set_path(ctx, path);
   gtk_style_context_set_screen (ctx, gtk_widget_get_screen(root_window));
 
-  GdkRGBA color, selected_color, bg_color, selected_bg_color;
-  gtk_style_context_get_color(ctx, GTK_STATE_FLAG_NORMAL, &color);
-  {
-    GdkRGBA *bc;
-    gtk_style_context_get(ctx, GTK_STATE_FLAG_NORMAL, "background-color", &bc, NULL);
-    bg_color = *bc;
-    gdk_rgba_free(bc);
-  }
-  gtk_style_context_get_color(ctx, GTK_STATE_FLAG_SELECTED, &selected_color);
-  {
-    GdkRGBA *sbc;
-    gtk_style_context_get(ctx, GTK_STATE_FLAG_SELECTED, "background-color", &sbc, NULL);
-    selected_bg_color = *sbc;
-    gdk_rgba_free(sbc);
-  }
+  GdkRGBA *tmpcolor;
+
+  // Get the normal foreground color from the CSS stylesheet
+  tmpcolor = g_malloc0(sizeof(GdkRGBA));
+  gdk_rgba_parse(tmpcolor, "#3C3C3C"); // .792f;
```

Comment:
`.792` is `CA`.

That being said, why do you allocate memory, parse the color, and then overwrite it with what we have in the CSS file? That will leak the `g_malloc0` result and be useless.

Reply:
The basic question here is: Can we guarantee that the CSS will always provide something? I'm not an expert on that. There were default values for these colors originally, although they were different than the default in the `darktable.css`:
```
darktable.bauhaus->bg_normal = 0.145098f;
darktable.bauhaus->bg_focus = 0.207843f;
darktable.bauhaus->text = .792f;
```
So at the end I decided to keep at least some defaults, in case the CSS does not provide. I can remove it easily, let me know.

Reply:
And I thought `gdk_rgba_free()` is freeing up the memory which has been allocated earlier. Is that a wrong assumption?

Reply:
That part is fine, you have to free what you get back from `gtk_style_context_get()`. But you will leak the `g_malloc0` result by using the same pointer.